src/codegen/templates/property_setter_template.py.mako

```diff
@@ -1,9 +1,13 @@
 <%def name="script_property_setter(attribute)">\
 <%
-        from codegen.utilities.attribute_helpers import get_generic_attribute_function_name, get_generic_attribute_function_type
+        from codegen.utilities.attribute_helpers import get_generic_attribute_function_name, get_generic_attribute_function_type, ATTRIBUTE_WITH_FILE_PATH_TYPE
     %>\
     @${attribute.name}.setter
+    %if attribute.name in ATTRIBUTE_WITH_FILE_PATH_TYPE:
+    def ${attribute.name}(self, val: Union[str, pathlib.PurePath]):
```

Comment:
If the property can return None, then it should accept None.
```suggestion
    def ${attribute.name}(self, val: Optional[Union[str, pathlib.PurePath]]):
```

Reply:
Noted, it's updated.